x-pack/plugins/enterprise_search/public/applications/shared/result/result_field.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import {
+  EuiCodeBlock,
+  EuiIcon,
+  EuiTableRow,
+  EuiTableRowCell,
+  EuiText,
+  EuiToken,
+} from '@elastic/eui';
+
+import { ResultFieldProps } from './types';
+import './result.scss';
+
+const iconMap: Record<string, string> = {
+  boolean: 'tokenBoolen',
+  date: 'tokenDate',
+  date_range: 'tokenDate',
+  double: 'tokenNumber',
+  double_range: 'tokenDate',
+  flattened: 'tokenObject',
+  float: 'tokenNumber',
+  float_range: 'tokenNumber',
+  geo_point: 'tokenGeo',
+  geo_shape: 'tokenGeo',
+  half_float: 'tokenNumber',
+  histogram: 'tokenHistogram',
+  integer: 'tokenNumber',
+  integer_range: 'tokenNumber',
+  ip: 'tokenIp',
+  ip_range: 'tokenIp',
+  join: 'tokenJoin',
+  keyword: 'tokenKeyword',
+  long: 'tokenNumber',
+  long_range: 'tokenNumber',
+  nested: 'tokenObject',
+  object: 'tokenObject',
+  percolator: 'tokenPercolator',
+  rank_feature: 'tokenRankFeature',
+  rank_features: 'tokenRankFeatures',
+  scaled_float: 'tokenNumber',
+  search_as_you_type: 'tokenSearchType',
+  shape: 'tokenShape',
+  short: 'tokenNumber',
+  text: 'tokenString',
+  token_count: 'tokenTokenCount',
+  unsigned_long: 'tokenNumber',
+};
+const defaultToken = 'questionInCircle';
+
+export const ResultField: React.FC<ResultFieldProps> = ({
+  iconType,
+  fieldName,
+  fieldValue,
+  fieldType,
+  isExpanded,
+}) => {
+  return (
+    <EuiTableRow className="resultField">
+      <EuiTableRowCell width="5%" valign="middle">
+        <span>
+          <EuiToken iconType={iconType || (fieldType ? iconMap[fieldType] : defaultToken)} />
+        </span>
+      </EuiTableRowCell>
+      <EuiTableRowCell width="25%" valign="middle">
+        <EuiText size="xs" grow={false}>
+          {fieldName}
+        </EuiText>
+      </EuiTableRowCell>
+      <EuiTableRowCell width="5%" valign="middle">
```

Comment:
```suggestion
      <EuiTableRowCell width="40" valign="middle">
```

Reply:
Ideally, `40` would be represented by an EUI variable... Not sure how to do that in JS, though. However, the hard-coded pixel value ensures nice spacing.